### PR TITLE
feat: Corestore.from(store)

### PR DESCRIPTION
**Pull request by Nuhvi**
Easy to do in app-land, but might be worth maintaining here for future internal changes.

### CODE REVIEW

**Comment by andrewosh** [comment]
Cool idea @Nazeh. Definitely want something like this, but any reason you prefer the static method vs. something like `store.session({ primaryKey })`, which `namespace` then calls internally (like you're doing with `from`)?

**Comment by Nuhvi** [comment]
Honestly no, I was just most recently working on Protomux, so that was my bias I guess 😄 

I will push a change with in a minute

File: README.md [hunk 1]
```diff
@@ -66,9 +66,8 @@ const core1 = ns1.get({ name: 'main' }) // These will load different Hypercores
 const core2 = ns2.get({ name: 'main' })
 ```
 
-#### `const storeB = Corestore.from(storeB, opts)`
-Create a new Corestore that shares resources with the original, like cache, sessions, replication streams, and storage, while reseting the namespace, and optionally overriding `primaryKey`.
-
+#### `const storeB = stroeA.session(opts)`
```
Comment (andrewosh) [inline README.md]: Typo "stroeA"

File: README.md [hunk 2]
```diff
@@ -66,9 +66,8 @@ const core1 = ns1.get({ name: 'main' }) // These will load different Hypercores
...
+Create a new Corestore that shares resources with the original, like cache, cores, replication streams, and storage, while reseting the namespace, and optionally overriding `primaryKey`.
```
Comment (andrewosh) [inline README.md]: reseting -> resetting

File: test/all.js [hunk 3]
```diff
@@ -341,41 +341,42 @@ test('core caching after reopen regression', async function (t) {
...
-  const store1 = Corestore.from(ns1, {
-    primaryKey: Buffer.alloc(32).fill('a')
-  })
-  const store2 = Corestore.from(ns2, {
-    primaryKey: Buffer.alloc(32).fill('a')
-  })
+  await store1.ready()
+  await store2.ready()
+
+  const session1 = store1.namespace('a').session()
```
Comment (andrewosh) [inline test/all.js]: Hmm, this line highlights a potential issue. It's reasonable to expect that calling `session` on a namespaced corestore returns a session with the same namespace by default, which would mean passing `namespace: this._namespace` as a constructor opt.

**Review by andrewosh — CHANGES_REQUESTED** [review]
Looks mostly good! Added a few nit typo comments and one point of confusion from the test.

**Comment by Nuhvi** [comment]
~A little problem with closing from a session if we reset the namespace, I guess there needs to be a flag like `this._isSession` if this to be merged at all! https://github.com/hypercore-protocol/corestore/blob/master/index.js#L300~ done at 6.0.5

**Comment by Nuhvi** [comment]
Updated this after the latest changes, ready for review.

**Review by andrewosh — APPROVED** [review]
LGTM

**Comment by andrewosh** [comment]
Published 6.2.0

**Pull request merged**